### DISCUSSION TITLE
Fatal error: require(): Failed opening required 'contrib/rsync.php'

### DISCUSSION THREAD

**Discussion by StijnCoolen** [deployphp/deployer#2931] [Help needed] [answered]
- Deployer version: 6.8

I have tried installing composer with both curl and composer. Both give the same error when trying to run `dep deploy` when I include `require contrib/rsync.php` in my deploy.php:

```
<?php
namespace Deployer;

require 'recipe/common.php';
require 'contrib/rsync.php';
```

**Comment by Schrank** [comment, marked as answer]
No clue why this doesn't work for your.

I'm running deployer as a phar file from [deployer.org](https://deployer.org/download) with this:

```
require 'contrib/slack.php';
```

And this doesn't throw an error. Are you sure you are running a version which includes `rsync.php`?

**Reply by StijnCoolen** [reply]
@Schrank I'm using the latest version of `deployer/dist` https://packagist.org/packages/deployer/dist, I also tried installing it with curl.

**Reply by antonmedv** [reply]
Probably you installed v6.

**Reply by StijnCoolen** [reply]
So using the following does not install latest?
```
curl -LO https://deployer.org/deployer.phar
mv deployer.phar /usr/local/bin/dep
chmod +x /usr/local/bin/dep
```
How do I install 7?

**Reply by antonmedv** [reply]
https://deployer.org/download

**Reply by StijnCoolen** [reply]
Thanks, I will try. Note that I also had the problem when I installed `7.0.0-rc.4` using composer.

**Reply by Schrank** [reply]
Hey! No https://deployer.org/deployer.phar is the latest stable and therefore 6.8.0.

I only run 7.0.0 phar, so can't help you with composer, but a (maybe unwanted - sorry!) recommendation: Don't run your build, deployment and testing tools from the same composer.json as your application. I recommend phar, so you are independent from all the dependencies phpunit, deployer, phpcs etc. bring you. And you can update your infrastructur stuff faster than your application, where the updates take longer - but the frameworks support older version longer as well :-)


Tested it and [7-rc4.phar](https://github.com/deployphp/deployer/releases/download/v7.0.0-rc.4/deployer.phar) works for me when living beside deploy.php with the content:

    <?php
    namespace Deployer;
    
    require 'recipe/common.php';
    require 'contrib/rsync.php';

Looks like this:

    -> % tree
    .
    ├── deploy.php
    └── deployer.phar

Hope that helps!

**Reply by StijnCoolen** [reply]
@Schrank how do I run the script from this .phar file? I have never used one before.
